src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -209,43 +209,51 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 {
                     Error(diagnostics, ErrorCode.ERR_AttributeCtorInParameter, node, attributeConstructor.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat));
                 }
+
+                //BindDefaultArguments(node, attributeConstructor.Parameters, analyzedArguments.ConstructorArguments.Arguments, analyzedArguments.ConstructorArguments.RefKinds, ref argsToParamsOpt, out defaultArguments, expanded, enableCallerInfo: true, diagnostics);
             }
 
             ImmutableArray<string?> boundConstructorArgumentNamesOpt = analyzedArguments.ConstructorArguments.GetNames();
             ImmutableArray<BoundAssignmentOperator> boundNamedArguments = analyzedArguments.NamedArguments?.ToImmutableAndFree() ?? ImmutableArray<BoundAssignmentOperator>.Empty;
             Debug.Assert(boundNamedArguments.All(arg => !arg.Right.NeedsToBeConverted()));
 
+            // var boundAttribute = new BoundAttribute(node, attributeConstructor, analyzedArguments.ConstructorArguments.Arguments.ToImmutableArray(), boundConstructorArgumentNamesOpt, argsToParamsOpt, expanded,
+            //     boundNamedArguments, resultKind, defaultArguments, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
             analyzedArguments.ConstructorArguments.Free();
-
+            //return boundAttribute;
+            var attributeData = GetAttribute(attributeType, attributeConstructor, boundConstructorArguments, boundNamedArguments, boundConstructorArgumentNamesOpt, argsToParamsOpt, node, hasErrors: resultKind != LookupResultKind.Viable, diagnostics);
             return new BoundAttribute(node, attributeConstructor, boundConstructorArguments, boundConstructorArgumentNamesOpt, argsToParamsOpt, expanded,
-                boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
+                boundNamedArguments, resultKind, attributeData, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
-        private CSharpAttributeData GetAttribute(BoundAttribute boundAttribute, BindingDiagnosticBag diagnostics)
+        internal CSharpAttributeData GetAttribute(
+            NamedTypeSymbol attributeType,
+            MethodSymbol? attributeConstructor,
+            ImmutableArray<BoundExpression> boundConstructorArguments,
+            ImmutableArray<BoundAssignmentOperator> boundNamedArguments,
+            ImmutableArray<string?> constructorArgumentNamesOpt,
+            ImmutableArray<int> constructorArgumentsToParamsOpt,
+            SyntaxNode node,
+            bool hasErrors,
+            BindingDiagnosticBag diagnostics)
```

Comment:
I made `CSharpAttributeData` part of the `BoundAttribute` node. So it needs to be computed *before* we create the `BoundAttribute`, hence changed the parameters as it cannot accept `BoundAttribute` now.